client/core/core.go

```diff
@@ -2676,7 +2676,6 @@ func (c *Core) authDEX(dc *dexConnection) error {
 		for _, match := range missing {
 			c.log.Warnf("DEX %s did not report active match %s on order %s - assuming revoked.",
 				dc.acct.host, match.id, oid)
-			match.failErr = fmt.Errorf("order not reported by the server on connect")
```

Comment:
@itswisdomagain had a footgun here.  Please have a look if you are able to.  These matches still needed recovery but failErr blocks them (e.g. isredeemable says no even if findmakerredemption finds it)

Reply:
Not sure I had an issue here, but looking at the `isRedeemable` checks, redeems will be skipped / ignored if failErr is set. Not setting failErr here is definitely a good thing, as it allows recovery actions to take place.

This PR also fixes the issue where findmakerredemption finds the counter-party's redemption but own redemption is not done because of a failErr set somewhere (e.g. here).  `findmakerredemption` now clears the failErr field to ensure `isRedeemable` returns true.

That said, I'd wanted to revisit the usages of the failErr property to be sure it won't cause unexpected issues. I generally doubt that we'd want to set a generic stop alert on matches in this way. Often-times, like in @jholdstock's scenario which is also documented in #361, we might want to permit retrials if an attempted action fails, rather than set a failErr property that pretty much says don't do anything further on this match. A quick look shows `match.failErr` is only checked in `isSwappable` and `isRedeemable`, but more importantly, what situations could cause us to not swap or redeem a match? Can we be more explicit about such conditions, using specific properties like `refundCoin` (which prevents redeeming) and `serverRevoked/selfRevoked` which prevents swapping?

Reply:
Thanks, @itswisdomagain.  That's our take too.  @buck54321 is digging in for improvements to the fail/retry possibilities you mentioned.  Clearly there are recoverable and/or transient errors that warrant trying repeatedly.  e.g. simnet harness has pushed wallet to limit sometimes.